cmd/extensions/main.go

```diff
@@ -170,6 +173,21 @@ func main() {
 		health = healthcheck.NewHandler()
 	}
 
+	podReady = true
+	health.AddReadinessCheck("agones-extensions", func() error {
+		if !podReady {
+			return errors.New("extensions Pod failed Readiness check, shutting down")
+		}
+		return nil
+	})
+
+	signals.NewSigTermHandler(func() {
```

Comment:
If we have a logger available, I would recommend adding a log message at the top and right before exiting. Something like "shutdown requested, failing readiness checks" at the top and "shutdown duration passed, exiting" or somesuch.

Reply:
The readiness check isn't the cause of the shutdown. Shutdown was requested (by `SIGTERM`), so we're failing readiness checks. Does that make sense? This log message is suggesting that failed readiness checks are somehow _why_ we're shutting down, but that's inverting the cause - instead, we're shutting down, so we're failing readiness checks to drain existing connections.